src/main/java/edu/cornell/kfs/module/ar/document/validation/impl/FirstInvoiceForAwardSuspensionCategory.java

```diff
@@ -0,0 +1,121 @@
+package edu.cornell.kfs.module.ar.document.validation.impl;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.kuali.kfs.coa.businessobject.Account;
+import org.kuali.kfs.integration.cg.ContractsAndGrantsBillingAward;
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.kfs.module.ar.ArConstants;
+import org.kuali.kfs.module.ar.ArPropertyConstants;
+import org.kuali.kfs.module.ar.businessobject.InvoiceAccountDetail;
+import org.kuali.kfs.module.ar.document.ContractsGrantsInvoiceDocument;
+import org.kuali.kfs.module.ar.document.dataaccess.ContractsGrantsInvoiceDocumentDao;
+import org.kuali.kfs.module.ar.document.validation.SuspensionCategoryBase;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.KFSPropertyConstants;
+import org.kuali.kfs.sys.document.service.FinancialSystemDocumentService;
+import org.kuali.rice.core.api.search.SearchOperator;
+
+import edu.cornell.kfs.module.ar.CuArPropertyConstants;
+
+public class FirstInvoiceForAwardSuspensionCategory extends SuspensionCategoryBase {
+
+    private FinancialSystemDocumentService financialSystemDocumentService;
+    private ContractsGrantsInvoiceDocumentDao contractsGrantsInvoiceDocumentDao;
+
+    @Override
+    public boolean shouldSuspend(ContractsGrantsInvoiceDocument contractsGrantsInvoiceDocument) {
+        return priorSuccessfulInvoicesDoNotExist(contractsGrantsInvoiceDocument);
+    }
+
+    protected boolean priorSuccessfulInvoicesDoNotExist(ContractsGrantsInvoiceDocument contractsGrantsInvoiceDocument) {
```

Comment:
This logic reads a little difficult to me.  Do you think it would be easier to read if this function were "priorSuccessfullInvoicesExist" . And then suspect only if that function is false?  I think it may be better to avoid the negative logic.  What do you think?

Reply:
Yes, I agree that this would make a little more sense as a positive-named method instead. I was initially trying to avoid the use of "!" in the shouldSuspend() method, but doing so may have made this more confusing. I'll update the name of this method and how it's used by shouldSuspend().